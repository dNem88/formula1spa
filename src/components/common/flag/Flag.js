import React from 'react'
import styles from './flag.module.css'
import flags from '../../../assets/images/flags/flag-images'

function Flag({imageBase, classname}) {
    
    return (
        <div className={styles[classname]}>
            <img src={flags[imageBase]} alt={`${imageBase}-flag`} loading='lazy'/>
        </div>
    )
}

export default Flag