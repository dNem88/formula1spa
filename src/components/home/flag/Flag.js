import React from 'react'
import styles from './flag.module.css'
function Flag(props) {
    // src={'../../../public/flags/props.flag.jpg'}
    return (
        <div className={styles['image-container']}>
            <img src={'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRKW5JRzY7qbTUAsJFK4ij7DPA7SDW7slWnvlyc7aY&usqp=CAU'} alt={'flag'}/>
        </div>
    )
}

export default Flag