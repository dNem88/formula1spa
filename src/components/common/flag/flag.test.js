import React from 'react'
import renderer from 'react-test-renderer'
import Flag from './Flag'
import {MemoryRouter} from 'react-router-dom'

let mock = {
    classname: 'test-class',
    imageBase: 'test-base'
}

describe('Test Flag', () => {
    it('Should Render Properly Flag', () => {
        let flag = renderer.create(
            <MemoryRouter>
                <Flag {...mock}/>
            </MemoryRouter>
        ).toJSON()
        expect(flag).toMatchSnapshot()
        expect(flag.type).toBe('div')
        expect(flag.children.length).toBe(1)
        expect(flag.props.className).toBe(mock.classname)
        expect(flag.children[0].type).toBe('img')
    })
})