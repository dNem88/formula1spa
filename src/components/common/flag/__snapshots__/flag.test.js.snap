// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Flag Should Render Properly Flag 1`] = `
<div
  className="test-class"
>
  <img
    alt="test-base-flag"
    loading="lazy"
  />
</div>
`;
